include/crow/http_server.h

```diff
@@ -175,21 +177,27 @@ namespace crow
         }
 
     private:
-        asio::io_service& pick_io_service()
+        int pick_io_service_idx()
         {
-            // TODO load balancing
-            roundrobin_index_++;
-            if (roundrobin_index_ >= io_service_pool_.size())
-                roundrobin_index_ = 0;
-            return *io_service_pool_[roundrobin_index_];
+            int min_queue_idx = 0;
+
+            // TODO improve load balancing
+            for (uint i = 1; i < task_queue_length_pool_.size(); i += 1)
+            {
+                if (task_queue_length_pool_[i] < task_queue_length_pool_[min_queue_idx])
+                    min_queue_idx = i;
+            }
+            return min_queue_idx;
         }
 
         void do_accept()
         {
-            asio::io_service& is = pick_io_service();
+            int service_idx = pick_io_service_idx();
+            asio::io_service& is = *io_service_pool_[service_idx];
             auto p = new Connection<Adaptor, Handler, Middlewares...>(
               is, handler_, server_name_, middlewares_,
-              get_cached_date_str_pool_[roundrobin_index_], *task_timer_pool_[roundrobin_index_], adaptor_ctx_);
+                get_cached_date_str_pool_[service_idx], *task_timer_pool_[service_idx], adaptor_ctx_, task_queue_length_pool_[service_idx]);
+            task_queue_length_pool_[service_idx] += 1;
```

Comment:
Aside from being able to use `task_queue_length_pool_[service_idx] ++`, it might be a good idea to have this line before initializing the connection.

Reply:
There's also a possible leak, if there's an ASIO error, the connection gets deleted but the counter isn't decremented. might be worth adding this below line `213`

Reply:
Ok, I moved up the incrementation.


Thanks, I didn't understand how this lambda function works, I added the decrementation now.